profile/filter.go

```diff
@@ -21,9 +21,19 @@ import "regexp"
 // FilterSamplesByName filters the samples in a profile and only keeps
 // samples where at least one frame matches focus but none match ignore.
 // Returns true is the corresponding regexp matched at least one sample.
-func (p *Profile) FilterSamplesByName(focus, ignore, hide, show *regexp.Regexp) (fm, im, hm, hnm bool) {
+func (p *Profile) FilterSamplesByName(focus, ignore, showFrom, hideFrom, show, hide *regexp.Regexp) (fm, im, sfm, hfm, sm, hm bool) {
```

Comment:
This function has become too complicated now. I would make it a separate function, something like ShowHideFrom and put it to prune.go, next to PruneFrom, since the functionality is rather related to pruning. Having the function separate would make the code and the tests more focused and easier to read.